### PR TITLE
Validate at least one release-note category is selected

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Validate a release note label is selected so we don't need to manually select them when updating `CHANGELOG.rst`.

## How is this patch tested?

A new github action

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
 ### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/release-note-label.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Validate
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { body, user } = context.payload.pull_request;
```
Comment (harupy) [inline .github/workflows/release-note-label.yml:23]: `body` represents a RR description.

File: .github/workflows/release-note-label.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+name: Release-note label
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
```
Comment (harupy) [inline .github/workflows/release-note-label.yml:11]: Add `edited` to support the following scenario:

- A user pushes a commit but no release-note label is selected -> check fails
- The user manually select one of release-note labels in the PR description -> check succeeds (without pushing another commit)

Comment (harupy) [reply .github/workflows/release-note-label.yml:11]: Without `edited`, we'd need to send another commit to trigger the check, which is obviously inconvenient.

File: .github/workflows/release-note-label.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            console.log("Selected release-note labels:");
+            console.log(selectedRNLabels);
+
+            // If the number of selected release-note labels is not 1, set the action status to "failed"
+            const numSelected = selectedRNLabels.length;
+            if (numSelected === 0) {
+              core.setFailed("No release-note label is selected. Please select one.");
+            } else if (numSelected > 1) {
+              core.setFailed("Multiple release-note labels are selected. Please select one.");
+            }
```
Comment (harupy) [inline .github/workflows/release-note-label.yml:62]: This shows an error message like this:

Screen Shot 2020-12-25 at 9 50 44

Comment (harupy) [reply .github/workflows/release-note-label.yml:62]: One might think "what is a release-note label?". One way to address this issue is:

Update the PR template as follows:

```diff
- ### Select release note label
+ <h3 id="release-note-category">Select release note label</h3>
```

This allows us jump to the release-note category section with an URL like `https://.../pull/xxx#release-note-category`.

File: .github/workflows/release_note_category.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+module.exports = ({ core, context }) => {
```
Comment (harupy) [inline .github/workflows/release_note_category.js:1]: Used a separate file to make it easier to review, debug, and auto-format.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**